app/controllers/api/v8/exercises_controller.rb

```diff
@@ -0,0 +1,93 @@
+class Api::V8::ExercisesController < Api::V8::BaseController
+  include Swagger::Blocks
+
+  swagger_path '/api/v8/courses/{course_id}/exercises' do
+    operation :get do
+      key :description, 'Returns all exercises of the course as json. Course is searched by id'
+      key :operationId, 'findExercisesById'
+      key :produces, ['application/json']
+      key :tags, ['exercise']
+      parameter '$ref': '#/parameters/path_course_id'
+      response 200 do
+        key :description, 'Exercises in json'
+        schema do
+          key :title, :exercises
+          key :required, [:exercises]
+          property :exercises do
+            key :type, :array
+            items do
+              key :'$ref', :ExerciseWithPoints
+            end
+          end
+        end
+      end
+      response 403, '$ref': '#/responses/auth_required'
+      response 404 do
+        key :description, 'Course not found'
+        schema do
+          key :title, :errors
+          key :type, :json
+        end
+      end
+    end
+  end
+
+  swagger_path '/api/v8/organizations/{organization_id}/courses/{course_name}/exercises' do
+    operation :get do
+      key :description, 'Returns all exercises of the course as json. Course is searched by name'
+      key :operationId, 'findExercisesByName'
+      key :produces, ['application/json']
+      key :tags, ['exercise']
+      parameter '$ref': '#/parameters/path_organization_id'
+      parameter '$ref': '#/parameters/path_course_name'
+      response 200 do
+        key :description, 'Exercises in json'
+        schema do
+          key :title, :exercises
+          key :required, [:exercises]
+          property :exercises do
+            key :type, :array
+            items do
+              key :'$ref', :ExerciseWithPoints
+            end
+          end
+        end
+      end
+      response 403, '$ref': '#/responses/auth_required'
+      response 404 do
+        key :description, 'Course or organization not found'
+        schema do
+          key :title, :errors
+          key :type, :json
+        end
+      end
+    end
+  end
+
+  def get_by_course
+    unauthorized_guest! if current_user.guest?
```

Comment:
I find the repeating work guest unnecessarily repetitive 
it could just be `unauthorized! if current_user.guest?`
or even 
`not_for_guests!` that handles the guest check in the method...